opc/resource_lbaas_policy.go

```diff
@@ -0,0 +1,901 @@
+		input.Type = "RateLimitingRequestPolicy"
+		input.RateLimitingRequestPolicyInfo = expandRateLimitingRequestPolicy(d)
+	}
+	if _, ok := d.GetOk("redirect_policy"); ok {
+		input.Type = "RedirectPolicy"
+		input.RedirectPolicyInfo = expandRedirectPolicy(d)
+	}
+	if _, ok := d.GetOk("resource_access_control_policy"); ok {
+		input.Type = "ResourceAccessControlPolicy"
+		input.ResourceAccessControlPolicyInfo = expandResourceAccessControlPolicy(d)
+	}
+	if _, ok := d.GetOk("set_request_header_policy"); ok {
+		input.Type = "SetRequestHeaderPolicy"
+		input.SetRequestHeaderPolicyInfo = expandSetRequestHeaderPolicy(d)
+	}
+	if _, ok := d.GetOk("ssl_negotiation_policy"); ok {
+		input.Type = "SSLNegotiationPolicy"
+		input.SSLNegotiationPolicyInfo = exapndSSLNegotiationPolicy(d)
+	}
+	if _, ok := d.GetOk("trusted_certificate_policy"); ok {
+		input.Type = "TrustedCertPolicy"
+		input.TrustedCertificatePolicyInfo = expandTrustedCertificatePolicy(d)
+	}
+
+	info, err := policyClient.CreatePolicy(lb, &input)
+	if err != nil {
+		return fmt.Errorf("Error creating Load Balancer Policy: %s", err)
+	}
+
+	d.SetId(fmt.Sprintf("%s/%s/%s", lb.Region, lb.Name, info.Name))
+	return resourcePolicyRead(d, meta)
+}
+
+func resourcePolicyRead(d *schema.ResourceData, meta interface{}) error {
+	lbaasClient, err := meta.(*Client).getLBaaSClient()
+	if err != nil {
+		return err
+	}
+	policyClient := lbaasClient.PolicyClient()
+	name := getLastNameInPath(d.Id())
+	lb := getLoadBalancerContextFromID(d.Id())
+
+	result, err := policyClient.GetPolicy(lb, name)
+	if err != nil {
+		// Policy does not exist
+		if client.WasNotFoundError(err) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error reading Load Balancer Policy %s: %s", d.Id(), err)
+	}
+
+	if result == nil {
+		d.SetId("")
+		return nil
+	}
+
+	d.Set("name", result.Name)
+	d.Set("state", result.State)
+	d.Set("type", result.Type)
+	d.Set("uri", result.URI)
+	d.Set("load_balancer", fmt.Sprintf("%s/%s", lb.Region, lb.Name))
+
+	if result.Type == "AppCookieStickinessPolicy" {
+		flattenApplicationCookieStickinessPolicy(d, result)
+	}
+	if result.Type == "CloudGatePolicy" {
+		flattenCloudGatePolicy(d, result)
+	}
+	if result.Type == "LBCookieStickinessPolicy" {
+		flattenLoadBalancerCookieStickinessPolicy(d, result)
+	}
+	if result.Type == "LoadBalancingMechanismPolicy" {
+		flattenLoadBalancingMechanismPolicy(d, result)
+	}
+	if result.Type == "RateLimitingRequestPolicy" {
+		flattenRateLimitingRequestPolicy(d, result)
+	}
+	if result.Type == "ResourceAccessControlPolicy" {
+		flattenResourceAccessControlPolicy(d, result)
+	}
+	if result.Type == "SetRequestHeaderPolicy" {
+		flattenSetRequestHeaderPolicy(d, result)
+	}
+	if result.Type == "SSLNegotiationPolicy" {
+		flattenSSLNegotiationPolicy(d, result)
+	}
+	if result.Type == "TrustedCertPolicy" {
+		flattenTrustedCertificatePolicy(d, result)
+	}
+
+	return nil
+}
+
+func resourcePolicyUpdate(d *schema.ResourceData, meta interface{}) error {
```

Comment:
What happens when you update to a different type? Let's add some tests around that and updating attributes of certain types

Reply:
added a couple of update tests